### PR TITLE
[nginx-ingress-controller] Allow custom health checks in upstreams

### LINKED ISSUE #818: Nginx Ingress Controller - Missing healthcheck params in upstream
Hi,

With current version (0.5) nginx marks a proxied server with fail status when a single request fails.

I think that the ningx template needs the healthcheck parameters for upstream section, like this:

upstream backend {
   server 10.1.0.102 max_fails=3 fail_timeout=30s;
}

### PR DESCRIPTION
**Pull request by aledbf**
fixes #818

### CODE REVIEW

**Comment by bprashanth** [comment]
still wip?

**Comment by aledbf** [comment]
Sorry. title updated

File: ingress/controllers/nginx/README.md [hunk 1]
```diff
@@ -196,6 +196,23 @@ Use the [custom-template](examples/custom-template/README.md) example as a guide
 **Please note the template is tied to the go code. Be sure to no change names in the variable `$cfg`**
 
 
+### Custom NGINX upstream checks
+
+NGINX exposes some flags in the [upstream configuration](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) that enabled configuration of each server in the upstream. The ingress controller allows custom `max_fails` and `fail_timeout` parameters in a global context using `upstream-max-fails` or `upstream-fail-timeout` in the NGINX Configmap or in a particular Ingress rule. By default this values are 0. This means NGINX will respect the `livenessProbe`, if is defined. If there is no probe, NGINX will not mark a server inside an upstream down.
```
Comment (bprashanth) [inline ingress/controllers/nginx/README.md]: s/liveness/readiness
Also this language makes me think nginx scrapes the pods for their readiness probes but that's not the case. Please clarify with something like: nginx will not health check your backends, and whenever the endpoints controller notices a readiness probe failure that pod's ip will be removed from the list of endpoints, causing nginx to also remove it from the upstreams.

**Comment by bprashanth** [comment]
LGTM but for the one comment, you can fix that in a follow up

**Pull request merged**